### PR TITLE
Convert CombineRgn to Cswin32 PInvoke

### PR DESCRIPTION
**Pull request by lonitra**
Utilize `PInvoke` for `CombineRgn` and add implicit conversion for `HRGN`

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7465)

### CODE REVIEW

**Comment by RussKie** [comment]
Please don't do this, rebase instead.

File: src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.RegionScope.cs [hunk 1]
```diff
@@ -117,7 +119,9 @@ public void Dispose()
                     DeleteObject(Region);
                 }
 
-                DisposalTracking.SuppressFinalize(this!);
+#if DEBUG
+                GC.SuppressFinalize(this);
+#endif
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.RegionScope.cs:124]: This isn't functionally equivalent. `DisposalTracking` throws if the instance isn't disposed corectly, i.e., there is a bug in our codebase. 
https://github.com/dotnet/winforms/blob/8f8a4a8976a0315e94b9738881d58881e7beac42/src/System.Windows.Forms.Primitives/src/System/Windows/Forms/DisposalTracking.cs#L40-L44

**Review by RussKie — DISMISSED** [review]

**Comment by lonitra** [comment]
Build Windows_x86 Debug failed with #7471

**Comment by JeremyKuhne** [comment]
> Please don't do this, rebase instead.

If this is possible with GitHub's conflict resolution, great. If it isn't I don't want to block being able to handle conflicts in NativeMethods.txt directly from the PR on GitHub.

**Comment by RussKie** [comment]
Rebase locally and f-push

**Pull request merged**